src/openrct2-ui/windows/Ride.cpp

```diff
@@ -3651,6 +3665,42 @@ static void window_ride_operating_update(rct_window* w)
     }
 }
 
+static void window_ride_operating_set_min_waiting_time(rct_window* w, uint8_t time)
+{
+    auto rideSetSetting = RideSetSettingAction(w->number, RideSetSetting::MinWaitingTime, time);
+    GameActions::Execute(&rideSetSetting);
+}
+
+static void window_ride_operating_set_max_waiting_time(rct_window* w, uint8_t time)
+{
+    auto rideSetSetting = RideSetSettingAction(w->number, RideSetSetting::MaxWaitingTime, time);
+    GameActions::Execute(&rideSetSetting);
+}
+
+static void window_ride_operating_textinput(rct_window* w, rct_widgetindex widgetIndex, char* text)
+{
+    if ((widgetIndex != WIDX_MINIMUM_LENGTH && widgetIndex != WIDX_MAXIMUM_LENGTH) || text == nullptr)
+        return;
+
+    int32_t time32;
+    char* end;
+
+    time32 = strtol(text, &end, 10);
+    if (*end == '\0')
+    {
+        uint8_t time = (uint8_t)std::clamp(time32, 0, 250);
```

Comment:
You need to use a C++-style cast.

But I'm a bit puzzled as to why you write to a `time32` and then cast to uint8_t in the first place.

Reply:
It was intended to make it so negative number input by the user would get turned into 0, but I suppose this case doesn't really matter